actions/register.ts

```diff
@@ -0,0 +1,62 @@
+import {
+  ActionFn,
+  Context,
+  Event,
+  TransactionEvent,
+  Storage,
+} from "@tenderly/actions";
+
+import { ethers } from "ethers";
+import { abi } from "./artifacts/ConditionalOrder.json";
+
+export const addContract: ActionFn = async (context: Context, event: Event) => {
+  const transactionEvent = event as TransactionEvent;
+  const iface = new ethers.utils.Interface(abi);
+
+  const registry = await Registry.load(context, transactionEvent.network);
+  console.log(`Current registry: ${JSON.stringify(registry.contracts)}`);
+
+  transactionEvent.logs.forEach((log) => {
+    if (log.topics[0] === iface.getEventTopic("ConditionalOrderCreated")) {
+      const contract = log.address;
+      if (
+        !registry.contracts.find((existing: string) => existing == contract)
```

Comment:
I think the "correct" condition here would be:
```suggestion
        registry.contracts.find((existing: string) => existing == contract) === undefined
```

Otherwise, some "falsy" conversion could potentially byte (hehe... pun intended) in the butt. Shouldn't be an issue here, but might as well protect against it.